src/components/DataPlatformProvider/DataPlatformProviderLoader.tsx

```diff
@@ -0,0 +1,235 @@
+      }),
+      url: JSONPath<string>({
+        json: data,
+        path: ['$', 'summary', 'gs_public_report'],
+        preventEval: true,
+        resultType: 'value',
+        wrap: false,
+      }),
+      values: mapPathsToValues([
+        ['$', 'summary', 'ein'],
+        ['$', 'summary', 'mission'],
+        ['$', 'summary', 'website_url'],
+        ['$', 'operations', 'blog_url'],
+        ['$', 'summary', 'telephone_numbers', '*', 'telephone_number'],
+        ['$', 'summary', 'social_media_urls', '*'],
+        ['$', 'summary', 'gs_profile_update_level'],
+        ['$', 'operations', 'leader_name'],
+        ['$', 'operations', 'leader_profile'],
+        ['$', 'operations', 'no_of_employees'],
+        ['$', 'operations', 'no_of_volunteers'],
+        ['$', 'operations', 'employees_greater_than_100K'],
+        ['$', 'operations', 'demographics', 'staff_level_totals', '*'],
+        ['$', 'operations', 'board_chair_name'],
+        ['$', 'operations', 'board_chair_affiliation'],
+        ['$', 'operations', 'board_of_directors', '*', 'name'],
+        ['$', 'summary', 'latitude'],
+        ['$', 'summary', 'longitude'],
+        ['$', 'summary', 'keywords'],
+        ['$', 'summary', 'ntee_code'],
+        ['$', 'summary', 'pcs_codes', '*', 'pcs_description'],
+        ['$', 'summary', 'profile_sdg_codes', '*', 'description'],
+        ['$', 'summary', 'addresses', '*', '*'],
+      ], data),
+    }),
+  },
+  'charity-navigator': {
+    name: 'Charity Navigator',
+    parse: (data: object) => ({
+      applicant: JSONPath<string>({
+        json: data,
+        path: ['$', 'title'],
+        preventEval: true,
+        resultType: 'value',
+        wrap: false,
+      }),
+      url: `https://www.charitynavigator.org${JSONPath<string>({
+        json: data,
+        path: ['$', 'url'],
+        preventEval: true,
+        resultType: 'value',
+        wrap: false,
+      })}`,
+      values: mapPathsToValues([
+        ['$', 'name'],
+        ['$', 'star_rating'],
+        ['$', 'ein'],
+        ['$', 'subsection'],
+        ['$', 'cause'],
+        ['$', 'city'],
+        ['$', 'state'],
+        ['$', 'size'],
+        ['$', 'acronym'],
+        ['$', 'donation_eligible'],
+        ['$', 'encompass_eligible'],
+        ['$', 'highest_level_advisory'],
+        ['$', 'akas', '*'],
+      ], data),
+    }),
+  },
+};
+type KnownProvider = keyof typeof providers;
+
+const isValidProvider = (provider: string): provider is KnownProvider => (
+  Object.hasOwn(providers, provider)
+);
+
+interface DataPlatformProviderProps {
+  externalId: string | undefined;
+  onClose: () => void;
+  provider: string;
+}
+
+const Invalid = ({
+  onClose,
+  provider,
+}: Omit<Required<DataPlatformProviderProps>, 'externalId'>) => (
+  <ClosablePanel
+    onClose={onClose}
+    title={(
+      <PanelTitle>
+        Invalid Provider
+      </PanelTitle>
+    )}
+  >
+    {`The data platform provider ${provider} is not recognized.`}
```

Comment:
This message - and the messages for the `<NoData>` and `<Loading>` components - is not padded correctly. @reefdog what should I be doing here - is there a wrapper or a class I should be using?

Reply:
There isn't, but there should be! This is one of the panel improvements I need to make. I'm going to try and do that as a separate PR that will cleanly merge with this one.